storage/storage.go

```diff
@@ -1838,6 +1841,33 @@ func applySourceConds(gen int64, conds *Conditions, call *raw.ObjectsRewriteCall
 	return nil
 }
 
+func applySourceCondsProto(gen int64, conds *Conditions, call *storagepb.RewriteObjectRequest) error {
```

Comment:
So odd that this requires a whole separate method when compose sources do not (seems like this is already the case in existing code though).

Reply:
Right there was already a source-specific conds helper for HTTP, so had to copy that for gRPC. Weird!